feat(discover): Add a toggle to hide prebuilt queries

wmak opened the pull request:
- This adds a toggle that lets a user hide prebuilt queries
- This is toggle is only enabled when there's at least 1 other query
  already so that they don't end up with an empty discover
<img width="1754" alt="image" src="https://user-images.githubusercontent.com/4205004/98956172-b8aa5b80-24cd-11eb-856e-cfee299fbc71.png">

wmak committed:
feat(discover): Add a toggle to hide prebuilt queries

- This adds a toggle that lets a user hide prebuilt queries
- This is toggle is only enabled when there's at least 1 other query
  already so that they don't end up with an empty discover

markstory commented on src/sentry/static/sentry/app/views/eventsV2/landing.tsx:
```diff
@@ -373,7 +408,7 @@ const StyledSearchBar = styled(SearchBar)`
   display: grid;
   grid-gap: ${space(2)};
-  grid-template-columns: auto min-content;
+  grid-template-columns: auto max-content min-content;
```
Does this need a different layout in mobile/tablet viewports?

wmak replied:
I made these stack instead on smaller viewports:
<img width="478" alt="image" src="https://user-images.githubusercontent.com/4205004/99104141-989e9900-25ae-11eb-98fb-5faa8a1c0019.png">

wmak committed:
ref: Changing layout in mobile

- Now the actions will stack instead of sliding off the page
- Also cleaning up an iffy if statement

Zylphrex commented on src/sentry/static/sentry/app/views/eventsV2/landing.tsx:
```diff
@@ -258,6 +267,15 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
+            isActive={renderPrebuilt}
+            isDisabled={renderPrebuilt && savedQueries.length === 0}
+            size="lg"
+            toggle={() => this.togglePrebuilt()}
```
```suggestion
            toggle={this.togglePrebuilt}
```

`togglePrebuilt` is already using an arrow function so `this` will be correctly bound. Using another arrow function here can cause more re-renders than necessary.

Zylphrex commented on src/sentry/static/sentry/app/views/eventsV2/landing.tsx:
```diff
@@ -274,6 +292,14 @@ class DiscoverLanding extends AsyncComponent<Props, State> {
+  togglePrebuilt = () => {
+    const {renderPrebuilt} = this.state;
+
+    setRenderPrebuilt(!renderPrebuilt);
```
Maybe this should be in the callback of `setState` once react finishes updating the state?

wmak committed:
ref: Addressing some PR comments

wmak committed:
Merge branch 'master' into feat/disable-prebuilt-queries
